coap-gateway/service/observation/resourcesObserver.go

```diff
@@ -83,55 +87,95 @@ func newResourcesObserver(deviceID string, coapConn ClientConn, callbacks Resour
 func (o *resourcesObserver) addResource(ctx context.Context, res *commands.Resource, obsInterface string) error {
 	o.lock.Lock()
 	defer o.lock.Unlock()
-	err := o.addResourceLocked(ctx, res, obsInterface)
+	obs, err := o.addResourceLocked(res, obsInterface)
+	if err == nil && obs != nil {
+		o.notifyAboutStartTwinSynchronization(ctx)
+		err = o.performObservationLocked([]*observedResource{obs})
+		if err != nil {
+			o.resources, _ = o.resources.removeByHref(obs.Href())
+			defer o.notifyAboutFinishTwinSynchronization(ctx)
```

Comment:
in case of success (err == nil) is there a different code path which emits this notification?

Reply:
Yes. In this case it will be emit by https://github.com/plgd-dev/hub/blob/jkralik/feature/reportsObservationsAreReady/coap-gateway/service/session.go#L466

Reply:
What if notifyContentChanged returns early with an error, won't the notification be missing then?

Reply:
In this case the notification `in sync` is not set for the device and this is true. User can try to fix it via reconnect device or set twin enabled to false and then back to true.